README.md

```diff
@@ -1,5 +1,109 @@
 
-1. [Week 1](./week1-webclient.md)
+## Introduction
+
+The course is going to be structured as a series of workshops, during which
+you will learn about various technologies and practices used in modern
+software development.
+
+In each workshop, you and your peers will be the development team working on
+part of a web application for a pet adoption center. You will learn about
+front-end web development, APIs, testing and continuous integration, as well
+as how to deploy web applications to the cloud, and how to collaborate in a
+modern software team.
+
+During the workshop, you will have access to a group of mentors, who will be
+there to help supplement the course materials and answer any questions you
+may have.
+
+## Learning goals
+
+Our goal is to give you a chance to learn about modern software development
+practices, and especially to learn about how web applications are designed,
+built, tested, and deployed.
+
+Over the duration of the course, you will build all the important parts of a
+web application that allows users to find and adopt adorable furry
+companions. Each workshop will focus on a different part of the application,
+and will draw on a different set of skills.
+
+Below, you'll find out what to expect in each workshop.
+
+## Course outline
+
+### Workshop 1: modern front-end web development
+
+In this workshop, we're going to build a frontend of a website for the pet
+adoption center. In doing this, you'll learn:
+
+- What is a single-page application (SPA)?
+- How to create a SPA using `create-react-app`.
+- How to scaffold a user interface using `react-bootstrap`.
+- What is an API (application programming interface)?
+- How to talk to an API from our React SPA.
+
+We'll also cover:
+
+- How to use the Git version control system to submit a pull request (PR).
+- How to use pull requests as part of a code review process.
+
+_If you're ready, go to the [course notes for Workshop 1](./week1-webclient.md)._
+
+### Workshop 2: APIs and back-end web development
+
+In the first workshop, you built a single-page application that talked to an
+API service that returned information about pets up for adoption. In this
+workshop, you're going to build the pet adoption API yourselves, and host it
+in the cloud. If things go according to plan, you'll learn:
+
+- How is an API application different from other web applications?
+- How to pick between different options for building and hosting web applications.
+- How to build an API for listing pets available for adoption.
+- How to host your API application in Azure.
+
+_The notes for Workshop 2 aren't quite ready yet. We'll add them here when
+they are._
+
+### Workshop 3: Testing, Continuous Integration, Continuous Deployment
+
+So far, you've built both a front-end and a back-end for the pet adoption
+center's website. In this workshop, we're going to look at how you can test
+the changes you make to your software, and how you can integrate those tests
```

Comment:
I'd hope that our participants have been writing tests all along and that testing isn't something that they tack-on afterwards. As such, I'd suggest phrasing this as specifically "how do we automate the running of tests by setting up an automated test runner on Azure, deploy automatically and how to map your release process to your Git process".

Reply:
Yup, that's fair. For now I'm not going to make explicit reference to testing in earlier sessions, but we can build it into the course materials, and I'll refocus this one to be about automation, CI, CD, etc.